servicenow-case-to-salesforce-case/main.bal

```diff
@@ -0,0 +1,80 @@
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mime;
+import ballerina/time;
+import ballerina/url;
+import ballerinax/salesforce as sf;
+
+configurable string servicenowInstance = ?;
+configurable string syncData = ?;
+configurable string serviceNowUsername = ?;
+configurable string serviceNowPassword = ?;
+configurable sf:ConnectionConfig salesforceConfig = ?;
+
+public function main() returns error? {
+    DateRange|error fetchPeriod = calculateFetchingPeriod();
+    if fetchPeriod is error {
+        log:printError("Error while calculating fetching period: ", fetchPeriod);
+        return;
+    }
+
+    CaseData[]|error cases = fetchCasesFromServiceNow(fetchPeriod.'start, fetchPeriod.end);
+    if cases is error {
+        log:printError("Error while fetching cases from ServiceNow: ", cases);
+        return;
+    }
+
+    error? e = addCasesToSalesforce(cases);
+    if e is error {
+        log:printError("Error while adding cases to Salesforce: ", e);
+        return;
+    }
+
+    check io:fileWriteString(syncData, check time:civilToString(fetchPeriod.now));
+    log:printInfo("Updated salesforce with cases from servicenow." +
+        " Cases added: " + cases.length().toString() +
+        " Update Timestamp: " + check time:civilToString(fetchPeriod.now));
+}
+
+function calculateFetchingPeriod() returns DateRange|error {
+    string lastFetchString = check io:fileReadString(syncData);
+    time:Civil lastFetch = check time:civilFromString(lastFetchString);
+    string 'start = string `'${lastFetch.year}-${lastFetch.month}-${lastFetch.day}','${lastFetch.hour}:${lastFetch.minute}:00'`;
```

Comment:
```suggestion
    string 'start = string `'${lastFetch.year}-${lastFetch.month}-${lastFetch.day}':'${lastFetch.hour}:${lastFetch.minute}:00'`;
```
This should be comma or a colon?

Reply:
This is the format expected by ServiceNow API.